dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/DefaultCustomIconService.java

```diff
@@ -55,170 +48,172 @@
  * @author Kristian Wærstad
  */
 @RequiredArgsConstructor
-@Service("org.hisp.dhis.icon.IconService")
-public class DefaultIconService implements IconService {
+@Service("org.hisp.dhis.icon.CustomIconService")
+public class DefaultCustomIconService implements CustomIconService {
+
+  private static final String CUSTOM_ICON_KEY_PATTERN = "^[a-zA-Z0-9*-_+]+$";
```

Comment:
If we have a pattern anyhow should we define a max length? Instead of `+` do a `{1,50}` or something? At the very least the `key` column will (or should) have a max length we can assume here.